src/settings/appsettings.cpp

```diff
@@ -986,6 +986,33 @@ bool AppSettings::defaultShowMagnifier()
     return false;
 }
 
+QMap<QString, QVariant> AppSettings::tesseractParameters() const
+{
+    QMap<QString, QVariant> parameters;
+    m_settings->beginGroup("Tesseract");
+    QStringList keys = m_settings->childKeys();
+    for (const auto &key : keys) {
+        parameters.insert(key, m_settings->value(key));
+    }
+    m_settings->endGroup();
+    return parameters;
+}
+
+void AppSettings::setTesseractParameters(const QMap<QString, QVariant> &parameters)
+{
+    m_settings->beginGroup("Tesseract");
+    m_settings->remove("");
```

Comment:
Settings can contain an empty keys?

Reply:
The API for settings groups is a little weird. Confusingly `clear()` always deletes all variables even inside a `beginGroup()`. Instead if you use `remove("")` inside a group it removes all the variables in the group. So the code is just replacing the whole `Tesseract` group with the values in `parameters`.

Reply:
Did not know about it :) Could you add a small comment about it?
```suggestion
    m_settings->remove({}); // Remove all keys in current group
```